extensions/amp-script/0.1/amp-script.js

```diff
@@ -563,11 +564,7 @@ export class SanitizerImpl {
     this.element_ = ampScript.element;
 
     /** @private @const {!Purifier} */
-    this.purifier_ = new Purifier(
-      ampScript.win.document,
-      dict({'IN_PLACE': true}),
-      rewriteAttributeValue
-    );
+    this.purifier_ = getServiceForDoc(this.element, 'purifier-inplace');
```

Comment:
Ohh, will not have an amp-doc yet. This should be delayed until necessary.

Reply:
So, should I be using `getServicePromiseForDoc` or `getServiceForDoc`?
It doesn't have many usages, i could convert all of them to async if necessary

Reply:
No, you just can't do it in the constructor. Use `buildCallback`. The problem isn't that the service isn't ready, it's that you attached the service to an ampdoc and we have no idea which ampdoc this element is supposed to use until it's in DOM. And it's not in DOM until `buildCallback`.

Reply:
Don't we have the window (and therefore `window.document` passed in the constructor)?
I guess...which window is passed in shadow mode?

Reply:
We could improve the whole situation further by removing `doc` from `Purifier` (right now its an ivar) and have it passed in during each call instead.

Reply:
Discussed offline. For `AmpMustache` there is not buildCallback since it extends from BaseTemplate instead of BaseElement. Went with converting the purifier into a window service that takes in `doc` for each request.